shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -894,6 +890,25 @@ protected FlutterEngine getFlutterEngine() {
     return delegate.getFlutterEngine();
   }
 
+  private Bundle cachedMetaData;
+
+  /** Mocks the meta data for testing purposes. */
+  @VisibleForTesting
+  public void setMetaData(Bundle metaData) {
+    cachedMetaData = metaData;
+  };
+
+  /** Retrieves the meta data specified in the AndroidManifest.xml. */
+  @Nullable
+  protected Bundle getMetaData() throws PackageManager.NameNotFoundException {
```

Comment:
The throw here is a bit burdensome for the subclasser. I would just re-throw it in a RuntimeException since it should be very unusual.